### ISSUE TITLE
influxDB URL in runtime preferences creates pattern based on Web UI protocol (HTTP or HTTPS), not choosen InfluxDB protocol

### ISSUE THREAD

**Issue by andpp** [ntop/ntopng#2453]
influxDB URL in runtime preferences always shows http**S** protocol even if I use http connection and 
redis-cli get "ntopng.prefs.ts_post_data_url" return "http://127.0.0.1:8086"

It is especially inconvenient when you want to enable/disable Layer-7 timeseries. Unless you change URL to correct protocol manually, it will switch to https.

Version	3.8.190307 - Community Edition
Built on	Ubuntu 18.04.1 LTS
nDPI	2.8.0-1455-1b2931e
cURL	7.58.0
 Twitter Bootstrap	3.x
 Font Awesome	4.x
RRDtool	1.7.0
InfluxDB	1.7.4
L7 Applications Resolution	10s
Redis Server	4.0.9
Mongoose web server	3.7
LuaJIT	Lua 5.3.5
ØMQ	4.2.5
GeoLite	1.3.1 
This product includes GeoLite data created by MaxMind.
Data-Driven Documents (d3js)	2.9.1 / 3.0

**Comment by emanuele-f** [comment]
Cannot reproduce this. I can successfully connect via http and the preference is properly set to http.

![2019-03-12_09-57](https://user-images.githubusercontent.com/5488003/54187157-d75c8180-44a4-11e9-80fd-f2a8e625b11d.png)

**Comment by andpp** [comment]
I tried to check the settings from another computer. The new computer shows correct value for InfluxDB URL. Looks like some issue with cache in browser. On my primary computer, showing error value, I use Chrome. On the second computer, showing correct value, I checked with Firefox. I'll check later on my primary computer if clearing of browser cache would help.

**Comment by andpp** [comment]
I checked it on Chrome. Clearing browser cache does not help.  There is the code from Chrome Code Inspect:

```
<td style="vertical-align:top; padding-left: 2em; white-space: nowrap;">
  <input id="id_input_ts_post_data_url" type="false" class="form-control" required=""
     pattern="^https?://.+$" spellcheck="false" name="ts_post_data_url" style="margin-bottom:0.5em;
     display:inline; margin-left:auto; text-align:right; width:20em;" value="https://127.0.0.1:8086">          </td>
```
I checked with firefox accessing to the web UI by HTTPS and got the same result. I guess **pattern="^https?://.+$"** is created based on the protocol I use to access to web UI.  In my previous message I accessed to web UI with HTTP protocol, not HTTPS.

I checked /usr/local/share/ntopng/scripts/lua/admin/prefs.lua:1255

```
prefsInputFieldPrefs(subpage_active.entries["influxdb_url"].title,
                       subpage_active.entries["influxdb_url"].description,
                       "ntopng.prefs.",
                       "ts_post_data_url",
                       "http://localhost:8086",
                       false, influx_active, nil, nil,  {attributes={spellcheck="false"}, pattern=getURLPattern(), required=true})
```

I guess getURLPattern() returns here an URL based on current protocol used for connection to the web UI. However the protocol for accessing to InfluxDB might be different

I pulled /ntopng/lua/admin/prefs.lua?tab=on_disk_ts page with http and https protocol. The only difference between pages is that ALL urls (even external) are correspond to the protocol I access to the page.  
For example for http protocol I got
```
<li><a href="http://blog.ntop.org/"
...
<h3 class="muted"><A href="/my/ntopng/"><svg width="103px" height="50px" viewBox="0 0 103 50"
  version="1.1" xmlns="http://www.w3.org/2000/svg"> 

```
for https protocol I got

```
<li><a href="https://blog.ntop.org/"
...
<h3 class="muted"><A href="/my/ntopng/"><svg width="103px" height="50px" viewBox="0 0 103 50"
  version="1.1" xmlns="https://www.w3.org/2000/svg"> 
```

I am  not sure if internal protocol should affect the protocol for external sites. I.e href="https://blog.ntop.org/" should always be http**S**.
I guess the same rule changes protocol in InfluxDB field

I use --http-prefix="/admin/lan/ntopng" so all internal URLs are showed correctly. For example href="/admin/lan/ntopng/css/flags.css"

**Comment by emanuele-f** [comment]
The pattern is always the same, see https://github.com/ntop/ntopng/blob/dev/scripts/lua/modules/lua_utils.lua#L2300 . If you have a problem with a specific browser (chrome) in the same ntopng installation, then it is probably a visualization problem of the browser itself (maybe it autocompletes the field), please post a screenshot

**Comment by andpp** [comment]
I found the root cause of the issue. My nginx server rewrites http:// to https:// for https access.

I did it because I use nginx as a proxy. I connect to nginx server using https and nginx connects to ntopng server using http so I got many complains from my browser about using together http and https URL in the same document because ntopng used absolute URL instead or relative. I guess it is fixed now.

This issue can be closed.
